codes/1963_cpp_wodnjs.cpp

```diff
@@ -0,0 +1,112 @@
+#include<iostream>
+#include<queue>
+#include<utility>
+
+using namespace std;
+bool not_prime[10000];
+int vis[10000];
+void prime_checker();
+int bfs(int from, int to);
+void clear_vis();
+int main() {
+	ios::sync_with_stdio(false);
+	cin.tie(nullptr);
+	cout.tie(nullptr);
+	int t_case;
+	cin >> t_case;
+	prime_checker();
+	for (int t = 0; t < t_case; t++)
+	{
+		int from, to;
+		cin >> from >> to;
+		int bfs_ret=bfs(from, to);
+		if(bfs_ret==-1)
+			cout << "Impossible\n";
+		else
+			cout << vis[to] - 1 << '\n';
+	}
+	return 0;
+}
+void prime_checker() {
+	for (int i = 2; i *i < 10000; i++)
+	{
+		if (not_prime[i] == false) 
+		{
+			for (int j = 2 * i; j < 10000; j += i)
```

Comment:
i * i로 고치면 사실상 같은 코드인데 더 효율적으로 작동합니다.
```suggestion
			for (int j = i * i; j < 10000; j += i)
```